bpf/bpf_overlay.c

```diff
@@ -345,7 +349,7 @@ static __always_inline int handle_ipv4(struct __ctx_buff *ctx,
 			if (!vtep)
 				goto skip_vtep;
 			if (vtep->tunnel_endpoint) {
-				if (*identity != WORLD_ID)
+				if (identity_is_world_ipv4(*identity))
```

Comment:
This looks like it reversed the logic, breaking VTEP support, see https://github.com/cilium/cilium/issues/31023.